Tests/ApolloTests/GETTransformerTests.swift

```diff
@@ -96,8 +225,69 @@ class GETTransformerTests: XCTestCase {
     let transformer = GraphQLGETTransformer(body: body, url: self.url)
     
     let url = transformer.createGetURL()
+    let queryString = url?.absoluteString == "http://localhost:8080/graphql?query=query%20HeroName($episode:%20Episode)%20%7B%0A%20%20hero(episode:%20$episode)%20%7B%0A%20%20%20%20__typename%0A%20%20%20%20name%0A%20%20%7D%0A%7D&variables=%7B%22episode%22:null%7D"
+    
+    XCTAssertTrue(queryString)
+  }
+  
+  func testEncodingQueryWith2DNullDefaultParameter() {
+    let operation = HeroNameQuery()
+    
+    let persistedQuery: GraphQLMap = [
+      "version": 1,
+      "sha256Hash": operation.operationIdentifier
+    ]
+    
+    let extensions: GraphQLMap = [
+      "persistedQuery": persistedQuery
+    ]
     
-    XCTAssertEqual(url?.absoluteString, "http://localhost:8080/graphql?query=query%20HeroName($episode:%20Episode)%20%7B%0A%20%20hero(episode:%20$episode)%20%7B%0A%20%20%20%20__typename%0A%20%20%20%20name%0A%20%20%7D%0A%7D&variables=%7B%22episode%22:null%7D")
+    let body: GraphQLMap = [
+      "query": operation.queryDocument,
+      "variables": operation.variables,
+      "extensions": extensions
+    ]
+    
+    let transformer = GraphQLGETTransformer(body: body, url: self.url)
+    
+    let url = transformer.createGetURL()
+    
+    if #available(iOS 11, macOS 13, watchOS 4, tvOS 11, *) {
+    let queryString = url?.absoluteString == "http://localhost:8080/graphql?extensions=%7B%22persistedQuery%22:%7B%22sha256Hash%22:%22f6e76545cd03aa21368d9969cb39447f6e836a16717823281803778e7805d671%22,%22version%22:1%7D%7D&query=query%20HeroName($episode:%20Episode)%20%7B%0A%20%20hero(episode:%20$episode)%20%7B%0A%20%20%20%20__typename%0A%20%20%20%20name%0A%20%20%7D%0A%7D&variables=%7B%22episode%22:null%7D"
+      XCTAssertTrue(queryString)
+    } else {
+      guard let variables = url?.queryItems?["variables"] else {
+        XCTFail("variables should not nil")
+        return
+      }
+      XCTAssertEqual(variables, "{\"episode\":null}")
```

Comment:
you can also check `variables.keys.contains("episode")` and then check that the value for that is null rather than checking the raw string.